### DISCUSSION TITLE
Ping Calculations: factor in interpolation buffer

### DISCUSSION THREAD

**Discussion by aaronkwan** [FirstGearGames/FishNet#731] [Feature Request]
o/

Looks like ping calculations don't factor in the interpolation buffer.

TimeManager, #region PingPong:
```C#
 internal void ModifyPing(uint clientTick)
 {
     uint tickDifference = (LocalTick - clientTick);
     byte interpBuffer = NetworkManager.PredictionManager.StateInterpolation; // something like this would work
     _pingAverage.ComputeAverage(tickDifference + interpBuffer);
     double averageInTime = (_pingAverage.Average * TickDelta * 1000);
     RoundTripTime = (long)Math.Round(averageInTime);
     _receivedPong = true;

     OnRoundTripTimeUpdated?.Invoke(RoundTripTime);
 }
```
Note that the pingdisplay component would need to correct for this if "Hide Tick Rate" is true.

**Comment by aaronkwan** [comment]
@FirstGearGames bump :)

**Comment by FirstGearGames** [comment]
the interpolation buffer does not affect the framework as a whole, just that feature. Adding that time duration to the ping would probably confuse a significant amount of people.

**Reply by aaronkwan** [reply]
true, but there's already an option to "Hide Tick Rate", which also does not affect the framework as a whole. maybe it could be a separate option or bundled into the above option?

**Reply by FirstGearGames** [reply]
> true, but there's already an option to "Hide Tick Rate", which also does not affect the framework as a whole. maybe it could be a separate option or bundled into the above option?

This is going a bit beyond the intentions of ping display. I'd recommend just copying the component and making your custom changes; the needed information is exposed.
